src/plugins/plugin-http.ts

```diff
@@ -44,118 +44,139 @@ function extractUrl(options) {
+          }
+          return on.apply(this, arguments);
+        };
+      });
+      res.on('end', function () {
+        requestLifecycleSpan
+          .addLabel(api.labels.HTTP_RESPONSE_SIZE_LABEL_KEY, numBytes);
+        requestLifecycleSpan
+          .addLabel(api.labels.HTTP_RESPONSE_CODE_LABEL_KEY, res.statusCode);
+        requestLifecycleSpan.endSpan();
+      });
+      if (callback) {
+        return callback(res);
+      }
+    });
+    api.wrapEmitter(req);
+    req.on('error', function (e) {
+      if (e) {
+        requestLifecycleSpan.addLabel(api.labels.ERROR_DETAILS_NAME, e.name);
+        requestLifecycleSpan
+          .addLabel(api.labels.ERROR_DETAILS_MESSAGE, e.message);
+      } else {
+        // What's the new logger target?
+        // console.error('HTTP request error was null or undefined', e);
+      }
+      requestLifecycleSpan.endSpan();
+    });
+    return req;
+  };
 }
 
 function patchHttp(http, api) {
   shimmer.wrap(http, 'request', function requestWrap(request) {
-    return makeRequestTrace(request);
+    return makeRequestTrace(request, api);
   });
 
   if (semver.satisfies(process.version, '>=8.0.0')) {
     // http.get in Node 8 calls the private copy of request rather than the one
     // we have patched on module.export. We need to patch get as well. Luckily,
     // the request patch we have does work for get as well.
     shimmer.wrap(http, 'get', function getWrap(get) {
-      return makeRequestTrace(get);
+      return makeRequestTrace(get, api);
     });
   }
+}
 
-  function isTraceAgentRequest (options) {
-    return options && options.headers &&
-      !!options.headers[api.constants.TRACE_AGENT_REQUEST_HEADER];
-  }
-
-  function makeRequestTrace(request) {
-    // On Node 8+ we use the following function to patch both request and get.
-    // Here `request` may also happen to be `get`.
-    return function request_trace(options, callback) {
-      if (!options) {
-        return request.apply(this, arguments);
-      }
-
-      // Don't trace ourselves lest we get into infinite loops
-      // Note: this would not be a problem if we guarantee buffering
-      // of trace api calls. If there is no buffering then each trace is
-      // an http call which will get a trace which will be an http call
-      if (isTraceAgentRequest(options)) {
-        return request.apply(this, arguments);
-      }
-
-      options = isString(options) ? url.parse(options) : merge({}, options);
-      options.headers = options.headers || {};
-
-      var uri = extractUrl(options);
-      var requestLifecycleSpan =
-          api.createChildSpan({name: getSpanName(options)});
-      if (!requestLifecycleSpan) {
-        return request.apply(this, arguments);
-      }
+function patchHttps(https, api) {
+  console.log('patch https');
+  shimmer.wrap(https, 'request', function requestWrap(request) {
+    return makeRequestTrace(request, api);
+  });
+  shimmer.wrap(https, 'get', function requestWrap(get) {
```

Comment:
`https.get` didn't exist before 8.0.0 as well right? I think this code assumes that you will never reach here because of the semver range in `module.exports` array. It would better to have the logic similar (potentially shared) between http and https rather than having two slightly different copies.

Reply:
I believe whether external/internal `request` is used for `get` is inconsistent between `http` and `https` across different versions of Node. AFAIK:

* `https.get` calls `https.request` for `<8.9.0` and (`>=8.9.1 && <9.0.0`)
* `http.get` calls `http.request` for `<8.0.0`

I think it's more readable to have slightly more duplicated code than have one patch and unpatch function and add an additional parameter that specifies the version range in which `get` must be separately patched/unpatched.